cmd/dagger/version.go

```diff
@@ -1,23 +1,59 @@
 package main
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
+	"io"
+	"os"
 	"runtime"
 
+	"dagger.io/dagger/telemetry"
+	"github.com/dagger/dagger/dagql/idtui"
+	"github.com/dagger/dagger/engine/distconsts"
+	enginetel "github.com/dagger/dagger/engine/telemetry"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/muesli/termenv"
+
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	"golang.org/x/mod/semver"
 
 	"github.com/dagger/dagger/engine"
 )
 
-var versionCmd = &cobra.Command{
-	Use:   "version",
-	Short: "Print dagger version",
-	// Disable version hook here to avoid double version check
-	PersistentPreRun: func(*cobra.Command, []string) {},
-	Args:             cobra.NoArgs,
-	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Fprintln(cmd.OutOrStdout(), long())
-	},
+var (
+	forceVersionCheck bool
+)
+
+func versionCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "version",
+		Short: "Print dagger version",
+		// Disable version hook here to avoid double version check
+		PersistentPreRun: func(*cobra.Command, []string) {},
+		Args:             cobra.NoArgs,
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Fprintln(cmd.OutOrStdout(), long())
+			if forceVersionCheck {
+				updateAvailable, err := updateAvailable(cmd.Context())
```

Comment:
I *think* this ends up getting the version twice potentially? Since it's called directly here, and indirectly from `checkForUpdates`?

Reply:
Ah woops my bad :cry: